### PR TITLE
Support expression evaluation inside of actors. 

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://75905336

This is a draft to help make sense of https://github.com/apple/swift/pull/37155. It needs some more work.

### CODE REVIEW

File: lldb/test/API/lang/swift/async/expr/main.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+actor Actor {
+  func f(_ n : Int) async {
+    f(0)     // Force a coro-split.
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/async/expr/main.swift]: Does this not require an `await`?

Why does this require a forced coroutine split?

Comment (adrian-prantl) [reply lldb/test/API/lang/swift/async/expr/main.swift]: It doesn't. Originally I thought a .resume function would be more interesting, but there isn't really a difference for this case. I'll remove it.

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp [hunk 2]
```diff
@@ -309,9 +309,14 @@ void SwiftASTManipulatorBase::DoInitialization() {
...
-  m_function_decl = func_finder.m_function_decl;
-  m_wrapper_decl = func_finder.m_wrapper_decl;
   m_extension_decl = func_finder.m_extension_decl;
+  if (m_extension_decl) {
+    m_function_decl = func_finder.m_function_decl;
+    m_wrapper_decl = func_finder.m_wrapper_decl;
+  } else {
+    m_function_decl = func_finder.m_wrapper_decl;
+    m_wrapper_decl = nullptr;
+  }
```
Comment (kastiglione) [inline lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp]: to verify my reading, this and the above changes are NFC, correct?

Comment (adrian-prantl) [reply lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp]: They correcting for the fact that now both lldb_expr and lldb_wrapped_expr have the @LLDBDebuggerFunction attribute. I don't like the implementation as is and I hope to update this with something better.

**Review by kastiglione — APPROVED** [review]

**Comment by adrian-prantl** [comment]
Updated patch is — I hope — much cleaner.

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/37155
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
Closing, this has been superceded by https://github.com/apple/llvm-project/pull/2904.

**Pull request closed without merging**